### PR TITLE
HPCC-14564 Disable -std=c++11 for eclrtl.cpp with g++ and clang due to boost (< 1.48) compile issue

### PR DESCRIPTION
**Pull request by mckellyln**
NOTE: this PR does not address issue of installation dependency on boost-regex version 1.46.1 when building with a newer version of boost-regex.

@xwang2713, @Michael-Gardner please review.

Signed-off-by: Mark Kelly mark.kelly@lexisnexis.com

### CODE REVIEW

File: rtl/eclrtl/CMakeLists.txt [hunk 1]
```diff
@@ -68,7 +66,15 @@ include_directories (
          ${BOOST_REGEX_INCLUDE_DIR}
     )
 
-ADD_DEFINITIONS( -D_USRDLL -DECLRTL_EXPORTS )
+if (UNIX AND USE_BOOST_REGEX)
+    if (EXISTS "/etc/debian_version")
+#       message("-- Boost_VERSION: ${Boost_VERSION}")
+        if (${Boost_VERSION} LESS 104800)
+            message("---- using special flag for some sources due to older boost version")
+            set_source_files_properties(eclrtl.cpp PROPERTIES COMPILE_FLAGS -std=c++98)
```
Comment (richardkchapman) [inline rtl/eclrtl/CMakeLists.txt]: Just make this line unconditional and all the concerns about trying to make the dependencies conditional go away...

Comment (mckellyln) [inline rtl/eclrtl/CMakeLists.txt]: ok, I reset flag if compiler is g++ or clang (which matches the conditions for when c++11 is set)

**Comment by ghalliday** [comment]
Note: We may need to split this file in two if we start using atomic classes in the jlib headers.

**Comment by Michael-Gardner** [comment]
A little late but I tested this and it all looks good.

**Pull request merged**